Unable to cast to CFString. Type ID: 0

I don't know if this is an issue with oshi or JNA, but...

OS: macOS Big Sur 11.4 Beta
JNA: 5.8.0
OSHI: 5.7.1
JVM: 11.0.11+9-LTS-194

```
Caused by: java.lang.ClassCastException: Unable to cast to CFString. Type ID: 0
	at com.sun.jna.platform.mac.CoreFoundation$CFStringRef.<init>(CoreFoundation.java:460)
	at oshi.software.os.mac.MacFileSystem.getFileStoreMatching(MacFileSystem.java:229)
	at oshi.software.os.mac.MacFileSystem.getFileStores(MacFileSystem.java:141)
	at oshi.software.common.AbstractFileSystem.getFileStores(AbstractFileSystem.java:55)
	at org.ziptie.zap.bootstrap.ZApplication.start(ZApplication.java:171)
```

The line of invocation looks like:
```java
171  systemInfo.getOperatingSystem().getFileSystem().getFileStores().stream()
172     .filter(fs -> "/data".equals(fs.getMount()))
173     .forEach(fs -> {
174        long totalSpace = fs.getTotalSpace();
175        long usableSpace = fs.getUsableSpace();
176        LOGGER.info("/data Disk Space (used/total): {}/{}", FormatUtil.formatBytes(totalSpace - usableSpace), FormatUtil.formatBytes(totalSpace));
177     });
```

Output of ``ls -l /Volumes``:
```
lrwxr-xr-x   1 root    wheel     1B Apr 24 03:09 Macintosh HD -> /
drwxrwxr-x@ 22 brettw  staff   816B Aug 11  2020 RAID1
drwxr-xr-x   1 brettw  staff   8.0K Apr 23 17:44 Untitled
drwxr-xr-x   3 root    wheel    96B May  2 04:39 com.apple.TimeMachine.localsnapshots
```

Output of ``mount``:
```
/dev/disk2s2s1 on / (apfs, sealed, local, read-only, journaled)
devfs on /dev (devfs, local, nobrowse)
/dev/disk2s5 on /System/Volumes/VM (apfs, local, noexec, journaled, noatime, nobrowse)
/dev/disk2s3 on /System/Volumes/Preboot (apfs, local, journaled, nobrowse)
/dev/disk2s6 on /System/Volumes/Update (apfs, local, journaled, nobrowse)
/dev/disk2s1 on /System/Volumes/Data (apfs, local, journaled, nobrowse)
/dev/disk3s3 on /Volumes/Untitled (ntfs, local, read-only, noowners)
map auto_home on /System/Volumes/Data/home (autofs, automounted, nobrowse)
/dev/disk5s2 on /Volumes/RAID1 (hfs, local, nodev, nosuid, journaled, noowners)
com.apple.TimeMachine.2021-05-02-033948.local@/dev/disk2s1 on /Volumes/com.apple.TimeMachine.localsnapshots/Backups.backupdb/iMac (2)/2021-05-02-033948/Macintosh HD - Data (apfs, local, read-only, journaled, nobrowse)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,8 @@
 ##### Bug fixes / Improvements
 * [#1614](https://github.com/oshi/oshi/pull/1614): Simplify Windows version parsing using JDK properties - [@dbwiddis](https://github.com/dbwiddis).
 * [#1620](https://github.com/oshi/oshi/pull/1620): Reduced log level to debug for fallback class loading - [@UnusualFrog](https://github.com/UnusualFrog).
+* [#1628](https://github.com/oshi/oshi/pull/1628): Null check volume name when iterating Mac File Stores - [@dbwiddis](https://github.com/dbwiddis).
+
 # 5.7.0 (2021-04-01), 5.7.1 (2021-04-15)
 
 ##### New Features

--- a/oshi-core/src/main/java/oshi/software/os/mac/MacFileSystem.java
+++ b/oshi-core/src/main/java/oshi/software/os/mac/MacFileSystem.java
@@ -226,9 +226,11 @@ public class MacFileSystem extends AbstractFileSystem {
                             if (diskInfo != null) {
                                 // get volume name from its key
                                 Pointer result = diskInfo.getValue(daVolumeNameKey);
-                                CFStringRef volumePtr = new CFStringRef(result);
-                                name = volumePtr.stringValue();
-                                if (name == null) {
+                                if (result != null) {
+                                    CFStringRef volumePtr = new CFStringRef(result);
+                                    name = volumePtr.stringValue();
+                                }
+                                if (name.isEmpty()) {
                                     name = Constants.UNKNOWN;
                                 }
                                 diskInfo.release();

